File: patch_server.py
with open("glue/server.ts", "r", encoding="utf-8") as f:
    code = f.read()

target = """  if ((res as any).artifacts) {
    advanced.artifacts = (res as any).artifacts;
  }"""

replacement = """  if ((res as any).artifacts) {
    advanced.artifacts = (res as any).artifacts;
  }
  if ((res as any).ciPassed === true) {
    advanced.ciPassed = true;
  }
  if ((res as any).reviewPassed === true) {
    advanced.reviewPassed = true;
  }"""

code = code.replace(target, replacement)

with open("glue/server.ts", "w", encoding="utf-8") as f:
    f.write(code)

print("patched")

File: patch_server.ts
import { readFile, writeFile } from "fs/promises";
async function run() {
  const file = "D:/clawWorld/jeo-claw/glue/server.ts";
  let content = await readFile(file, "utf8");
  content = content.replace(
    /await dispatchStageWork\(state, \{([\s\S]*?)\}\);\s+const advanced = advanceStage\(state\);\s+return advanced;/m,
    `const res = await dispatchStageWork(state, {$1});\n  const advanced = advanceStage(state);\n  if ((res as any).artifacts) {\n    advanced.artifacts = (res as any).artifacts;\n  }\n  return advanced;`
  );
  await writeFile(file, content, "utf8");
  console.log("Patched server.ts!");
}
run();
